carma-messenger-core/cpp_message/src/cpp_message.cpp

```diff
@@ -126,32 +126,29 @@ namespace cpp_message
         message->value.choice.TestMessage04.body.scale = msg->scale;
         // copy bounds
         auto count = msg->bounds.size();
-        if(count > 0)
-        {
-            for(auto i = 0; i < count; i++) {
-                // construct control bounds
-                ControlBounds_t* bounds_p;
-                bounds_p = (ControlBounds_t*) calloc(1, sizeof(ControlBounds_t));
-                bounds_p->lat = msg->bounds[i].latitude;
-                bounds_p->lon = msg->bounds[i].longitude;
-                // copy offsets from array to C list struct
-                auto offset_count = msg->bounds[i].offsets.size();
-                if(offset_count > 0) {
-                    for(auto j = 0; j < offset_count; j++) {
-                        int16_t temp = msg->bounds[i].offsets[j];
-                        asn_sequence_add(&bounds_p->offsets.list, &temp);
-                    }
+        for(auto i = 0; i < count; i++) {
+            // construct control bounds
+            ControlBounds_t* bounds_p;
+            bounds_p = (ControlBounds_t*) calloc(1, sizeof(ControlBounds_t));
+            bounds_p->lat = msg->bounds[i].latitude;
+            bounds_p->lon = msg->bounds[i].longitude;
+            // copy offsets from array to C list struct
+            auto offset_count = msg->bounds[i].offsets.size();
+            if(offset_count > 0) {
+                for(auto j = 0; j < offset_count; j++) {
+                    int16_t temp = msg->bounds[i].offsets[j];
+                    asn_sequence_add(&bounds_p->offsets.list, &temp);
                 }
-                // convert a long value to an 8-bit array of length 8
-                uint8_t oldest_val[8];
-                for(auto k = 7; k >= 0; k--) {
-                    // TODO this line needs to be tested
-                    oldest_val[7 - k] = msg->bounds[i].oldest >> (k * 8);
-                }
-                bounds_p->oldest.size = 8;
-                bounds_p->oldest.buf = oldest_val;
-                asn_sequence_add(&message->value.choice.TestMessage04.body.bounds.list, bounds_p);   
             }
+            // convert a long value to an 8-bit array of length 8
+            uint8_t oldest_val[8];
+            for(auto k = 7; k >= 0; k--) {
+                // TODO this line needs to be tested
```

Comment:
Does this TODO comment still apply?

Reply:
Yup. I am building a test case for that.